### PR TITLE
Update the multigate example (add performance and remove dwavebinarycsp)

### PR DESCRIPTION
**Pull request by JoelPasvolsky**

### CODE REVIEW

File: docs/examples/multi_gate.rst
Comment (arcondello) [inline docs/examples/multi_gate.rst:111]: why not name these functionally? Like `or_bc = or_gate('b', 'c', 'out2')` or whatever

File: docs/examples/multi_gate.rst
Comment (arcondello) [inline docs/examples/multi_gate.rst:132]: Why not just `print(solutions)` and use the pretty printing?

File: docs/examples/multi_gate.rst
Comment (hhtong) [inline docs/examples/multi_gate.rst:594]: `sampler2` isn't defined here if `sampleset.first.energy > 0`

File: docs/examples/multi_gate.rst
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:478]: Rewrote, thanks

File: docs/examples/multi_gate.rst
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:132]: The internal variables double the variable count, so it's harder to see ``z=func(a, b, c, d)`` in the output

File: docs/examples/multi_gate.rst
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:111]: It makes an easy connection between the BQM for a gate and that gate's numbering in the figure above it (e.g., the users sees "bqm4" and then looks up the gate with a big "4" on it in the figure) 
The functionality is easily readable from the rhs (operation in the gate name and inputs in the arguments) so adding those to the lhs wouldn't add helpful info. 
If this content moves in the future somewhere that separates code from description (e.g., an example with a readme or JN) then functional names would be better, but close on the same page, this makes a nice connection

Comment (arcondello) [reply docs/examples/multi_gate.rst:111]: Sure, defer to what you think makes sense. Though IMO if that's the reasoning I might name them `gate4 = or_gate(...)`

File: docs/examples/multi_gate.rst
Comment (arcondello) [reply docs/examples/multi_gate.rst:132]: A good future use case for https://github.com/dwavesystems/dimod/pull/806 though IMO

Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:132]: Oh, that's a nice PR

File: docs/examples/multi_gate.rst
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:111]: Sure, I updated to ``bqm_gate<num gate>`` (I want to also keep the "bqm" to emphasize that they are creating BQMs and then adding them to a comprehensive BQM

File: docs/examples/multi_gate.rst
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst]: Reimplemented using https://github.com/dwavesystems/dimod/pull/806 -- anything else needed for approval & merge?

**Review by randomir — APPROVED** [review]
Lovely.

**Review by arcondello — APPROVED** [review]

**Comment by randomir** [comment]
@JoelPasvolsky, for doctest to succeed, you'll need to require `dimod>=0.10.9` in test or docs requirements.

File: docs/examples/multi_gate.rst
Comment (JoelPasvolsky) [reply docs/examples/multi_gate.rst:457]: Removed for this instance

**Comment by JoelPasvolsky** [comment]
CI is failing for https://github.com/dwavesystems/dwave-ocean-sdk/pull/168 on a dwave-hybrid unittest

**Pull request merged**
